CommonTools/TriggerUtils/src/GenericTriggerEventFlag.cc

```diff
@@ -278,71 +282,132 @@ bool GenericTriggerEventFlag::acceptDcs(const edm::Event& event) {
   if (!onDcs_ || dcsPartitions_.empty())
     return (!andOr_);  // logically neutral, depending on base logical connective
 
+  bool useDCSRecord(false);
+
   // Accessing the DcsStatusCollection
   edm::Handle<DcsStatusCollection> dcsStatus;
   event.getByToken(dcsInputToken_, dcsStatus);
-  if (!dcsStatus.isValid()) {
+
+  edm::Handle<DCSRecord> dcsRecord;
+  event.getByToken(dcsRecordToken_, dcsRecord);
+
+  if (!dcsStatus.isValid() && !dcsRecord.isValid()) {
     if (verbose_ > 1)
       edm::LogWarning("GenericTriggerEventFlag")
-          << "DcsStatusCollection product with InputTag \"" << dcsInputTag_.encode()
-          << "\" not in event ==> decision: " << errorReplyDcs_;
+          << "DcsStatusCollection product with InputTag \"" << dcsInputTag_.encode() << "\" not in event \n"
+          << "DCSRecord product with InputTag \"" << dcsRecordInputTag_.encode() << "\" not in event \n"
+          << " ==> decision: " << errorReplyDcs_;
     return errorReplyDcs_;
   }
   if ((*dcsStatus).empty()) {
```

Comment:
The check in L294 takes care of when both handles are invalid, but it looks like we can get to L302 even if `dcsStatus.isValid()==false` (and I guess that would lead to an exception). I'm not very familiar with these DCS classes, but maybe it's enough to change `(*dcsStatus).empty()` to `!dcsStatus.isValid() or dcsStatus->empty()` ?

Reply:
@missirol, thanks Marino, I'll update the code according to the suggestion, but I would prefer to touch again after https://github.com/cms-sw/cmssw/pull/37083 is merged (and this one is rebased and changed accordingly).

Reply:
Sure, thanks!